packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -695,8 +702,31 @@ export class Combobox
     if (this.ignoreSelectedEventsFlag) {
       return;
     }
-
     const target = event.target as HTMLCalciteComboboxItemElement["el"];
+
+    const isSelectAllTarget = event
+      .composedPath()
+      .some((node) => (node as HTMLElement) === this.selectAllComboboxItemReferenceEl);
+
+    if (this.selectAllEnabled) {
+      if (isSelectAllTarget) {
+        this.toggleSelectAll();
+      }
+      this.updateAndGetSelectAllState();
+
+      // TODO: Doesn't hide the last element
+      if (this.isAllSelected()) {
```

Comment:
Can we open up a refactor issue for combobox to use more state for items and whether they are all selected? Ideally, there should just be a stateful value for whether all items are selected instead of having to call a method.

Reply:
[Refactor calcite-combobox to Use Stateful Properties](https://github.com/Esri/calcite-design-system/issues/12061#top)#12061